docs/_spec/08-pattern-matching.md

```diff
@@ -307,6 +309,46 @@ All alternative patterns are type checked with the expected type of the pattern.
 They may not bind variables other than wildcards.
 The alternative pattern matches a value ´v´ if at least one its alternatives matches ´v´.
 
+### Given patterns
+
+```ebnf
+  GivenPattern   ::= TypePat
+```
+
+A _given pattern_ introduces an _anonymous given instance_ of the _type_ matched by `TypePat` to the scope of the clause that a given pattern occurs _within_.
+
+The rules for the _name_ synthesized for each anonymous instance is defined in the reference.
+
+If multiple names synthesized by the compiler are _not_ unique, the pattern is [illegal](#bound-variables).
+
+###### Example 1
+
+In the following example, the type pattern matches a non-null instance of the `String` class. 
+The given pattern _introduces_ an _anonymous given instance_ of type `String`, synthesized as `given_String`, to the local scope. 
+
+Summoning an implicit of type `String` within the body of the case clause resolves to this new instance; the program prints `"bar"` instead of `"foo"` first.
+
+```scala
+
+given String = "foo"
+
+"bar" match
+  case given String => println(summon[String]) // "bar" is printed
+
+println(summon[String]) // "foo" is printed
+```
+###### Example 2
+
+In the following example, a generic type pattern `List[_]` is bound. 
+This matches a type of `List[t]` where `t` is an anonymous type _within_ the local scope. 
+`t` is _inferred_ to be a subtype of `Int` within the body of the case clause and can return `Int` without any casts.
+
+```scala
+def foo: Int =
+  List(1) match
+    case given List[_] => summon[List[?]].head // returns `1`
```

Comment:
This example does not illustrate that we statically receive an `Int`. Include a `val x: Int = summon[List[Int]].head`?

Reply:
Sure, the return type is `Int` of the function, but I can introduce a `val` if it would make it clearer.

Reply:
scala 2
```
scala> List(42) match { case xs @ (_: List[_]) => xs }
val res1: List[Any] = List(42)
```
scala 3
```
scala> List(42) match { case xs @ (_: List[_]) => xs }
val res0: List[Int] = List(42)
```
I'm not sure if scala 2 is plain wrong here. So any clarification would be helpful.
Oh nm it's REPL output bc:
```
scala> val ys: List[Int] = List(42) match { case xs @ (_: List[_]) => xs }
val ys: List[Int] = List(42)
```
but that goes to my point that a) a semi-casual reader might not have a preconceived notion and b) if I try it in the REPL, REPL may let me down, so I rely on your examples before giving up.

Reply:
> Sure, the return type is `Int` of the function, but I can introduce a `val` if it would make it clearer.

The reader of the spec does know that, and does not have an IDE or REPL to prove it. So yes, a `val` with an expected type would make it a lot clearer.